### ISSUE TITLE
[BOUNTY] update ReLoRA implementation

### ISSUE THREAD

**Issue by winglian** [axolotl-ai-cloud/axolotl#1198]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

relora improved their implementation as the original wasn't very good. 

Value of the bounty is $100 in runpod GPU credits. 

### ✔️ Solution

see https://github.com/Guitaricet/relora/commit/d8ff7ee471b16bba480bf28c92b00e78b91a8cb2#diff-0d94c276364524bc8e6bdd0d43b1ee104fe56a541129a1e588aeffd87bb22840

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by HarjobandeepSingh** [comment]
can you assign me plz @winglian

**Comment by SwapnilDreams100** [comment]
Hi @winglian, happy to start working on this. Are there any specific changes you are looking for? It seems like the key changes implemented in the new ReLora implementation is the addition of quantization, which is already present in the current implementation.

**Comment by ehartford** [comment]
This is exciting, I was pretty sad when the original ReLoRA didn't work out.  Would love to do some pretraining on a GPU-poor setup!

**Comment by aeryncaen** [comment]
I'd like to add an additional bounty to this of $400 in credits at runpod. I pretty much /need/ this in order to train some models.

**Comment by winglian** [comment]
Going to close this. I've added magnitude pruning of the optimizer states in https://github.com/OpenAccess-AI-Collective/axolotl/pull/1245 which is the main change
